### DISCUSSION TITLE
SQL Anywhere adapter

### DISCUSSION THREAD

**Discussion by glennfu** [vcr/vcr#876] [Ideas]
Hello there! I've got a project interacting with a SQL Anywhere AR adapter, which means my database.yml has a `commlinks: TCP(HOST=0.some.domain.here:12345)` or `commlinks: TCPIP(HOST=0.some.domain.here:12345)` in it.

The database on the other side, being a bit of a black box, definitely needs some good test coverage, so I don't want to go around stubbing everything out. Also, there's a fair bit of lag/latency. Exactly for the same reasons I would want to use VCR on an HTTP API, I want to use VCR on this.

Basically, the cassettes should have the SQL string in place of uri/method/body/headers for the request, and the response would just be whatever gets returned I guess.

Here's SQL Anywhere: https://github.com/sqlanywhere/sqlanywhere

Has anyone found/used/needed something like this before? If I'm going to build this myself, can anyone give me a pointer into the right direction?

**Comment by krainboltgreene** [comment]
We've discussed internally the idea of VCR handling any network request. It's a rather large task, though.
